### DISCUSSION TITLE
How to use this in monorepo setup?

### DISCUSSION THREAD

**Discussion by whalemare** [ivanhofer/typesafe-i18n#656] [Q&A] [answered]
I have monorepo (nx.dev) with next folder structure:

```
--apps
----app_one
----app_two
--libs
----lib_three
----lib_four
```

I want to separate my locales from each other (each app and each library should have it own locales).
After some research I realize, that it's impossible because of

https://github.com/ivanhofer/typesafe-i18n/blob/38a9c97520efa1a6fc603cd8b83d71bd81f0e0b2/packages/config/src/core.mts#L27

that hardcodes `option` file inside code.

### Possible usage
I want to use this library, like `tsc` API does, by specify required project declaration `tsc -p tsconfig.lib.json`

For example, have separated `typesafe-18n.*.json` files that store configuration for each project:

_typesafe-18n.app_one.json_
```json
{
   "$schema": "https://unpkg.com/typesafe-i18n@5.18.1/schema/typesafe-i18n.json",
   "baseLocale": "en",
   "adapter": "react",
   "generateOnlyTypes": true,
   "outputPath": "./apps/app_one/src/internal/locale"
}
```

_typesafe-18n.lib_three.json_
```json
{
   "$schema": "https://unpkg.com/typesafe-i18n@5.18.1/schema/typesafe-i18n.json",
   "baseLocale": "en",
   "adapter": "react",
   "generateOnlyTypes": true,
   "outputPath": "./libs/lib_three/src/internal/locale"
}
```

and then call regeneration for each like:

```bash
yarn typesafe-i18n -p typesafe-18n.app_one.json # generate locales inside apps/app_one
yarn typesafe-i18n -p typesafe-18n.lib_three.json # generate locales inside libs/lib_three
```

Maybe I am wrong and should to do it in another way? But, unfortunately, unable to find it

**Comment by ivanhofer** [comment, marked as answer]
Hi @whalemare It is currently not possible to specify a file path to read the config. Feel free to crete a PR if you have time. I would love to add that feature, but I currently don't have the resources to implement it.

As a workaround you can create the config files in different folders, and set the "current working directory (cwd)" to that folder and then execute the cli. Or simply "cd" into that folder and execute it from there: `cd app_one && yarn typesafe-i18n`

**Reply by whalemare** [reply]
Okay, I will try to make PR
